### DISCUSSION TITLE
Setting label from model data on Resource pages

### DISCUSSION THREAD

**Discussion by happyclicker** [filamentphp/filament#10632] [Help] [answered]
I discovered I can call this function (v3.x):

`public static function getModelLabel(): string
    {
        return 'Oink';
    }`

And when I view the resource (Infolist, or Edit), I get "View Oink" or "Edit Oink".  

What I would like to do is grab a piece of the model data (such as the model field "sku") and do something like this:

`public static function getModelLabel(): string
    {
        return $model->sku;
    }`

I just don't see where the model data is exposed to grab this field?  I am having a hard time fully navigating the docs, things are not organized in a way that is easy (for me, anyway) - but I am a novice at this package, so it's a bit early for me to try to contribute to help make things more clear.

**Comment by leandrocfe** [comment, marked as answer]
Use page title in the EditPage and ViewPage

https://filamentphp.com/docs/3.x/panels/pages/#customizing-the-page-title

You can use `$this->getRecord()->sku` to get the data

**Reply by happyclicker** [reply]
Ah!  I was trying to do this in the main Resource file, which of course would never work.  

Thank you for pointing me to the "$this->getRecord()", I searched specifically for that mention, and it is in the Custom Columns section for anyone else looking.

[https://filamentphp.com/docs/3.x/tables/columns/custom#accessing-the-eloquent-record](https://filamentphp.com/docs/3.x/tables/columns/custom#accessing-the-eloquent-record)
